### DISCUSSION TITLE
[Statiq.Docs] How to exclude Docable namespace from docs generation?

### DISCUSSION THREAD

**Discussion by fauxnik** [statiqdev/Discussions#207] [Q&A]
I'm new to Statiq.Docs and just getting started. After some initial confusion, I finally figured out that a theme was required for automatic doc site generation, and I added Docable as a git submodule.

Now the API docs for my code are being generated, but I see they also include the `Docable` namespace. In trying to remove this, I've changed the `SourceFiles` and `Excluded` settings to no avail.

How might I exclude the `Docable` namespace from the generated API site?

**Comment by TheBrambleShark** [comment]
Where does the code for your docable site live? If you place it under `sitedir/theme` it should be treated as compiled output rather than as source documents.

Default globbing pattern on the [Specifying Code](https://www.statiq.dev/guide/docs/specifying-code) page indicates that it looks for source code under the `sitedir/src` and `sitedir/../src` directories.

**Reply by daveaglick** [reply]
Yep, this is exactly right. If the Docable code itself is being included, I'm curious where it's located directory-wise. Can you let us know what your full path structure looks like (where is Docable code, where is your own API code, etc.)?

**Reply by fauxnik** [reply]
Here's an outline of my project directory structure:

```
project_root
├─ CommsRadioAPI
│  └─ my api code...
└─ CommsRadioDocsSite
   └─ theme
      └─ Docable code...
```

I was trying to configure the settings file such that, were I to add any other directories in the future, they'd get picked up automatically.

```yaml
SourceFiles:
  - ../../{!CommsRadioDocsSite}/**/{!.git,!bin,!obj,!packages,!*.Tests,}/**/*.cs
```

I don't really need to do this, so I fixed it for myself by changing the settings file like so:

```yaml
SourceFiles:
  - ../../CommsRadioAPI/**/{!.git,!bin,!obj,!packages,!*.Tests,}/**/*.cs
```
